packages/knit-gateway/src/schema.ts

```diff
@@ -146,43 +174,50 @@ function getMessageSchema(message: MessageType): ComputedSchema {
   // To handle recursive types, we need to add the schema to the table before
   // computing the fields.
   messageToSchemaTable.set(message.typeName, schema);
-  const fields = computeMessageFields(message);
+  const fields = computeMessageFields(message, relations);
   schema.fields = fields;
   schema.localNameTable = new Map(fields.map((f) => [f.name, f]));
   return schema;
 }
 
 function computeMessageFields(
-  message: MessageType
-): PlainMessage<Schema_Field>[] {
+  message: MessageType,
+  relations: RelationsMap
+): PlainSchemaField[] {
   if (wktSet.has(message.typeName)) {
     return [];
   }
-  let fields: PlainMessage<Schema_Field>[] = [];
+  const fields: PlainSchemaField[] = [];
   for (const protoField of message.fields.list()) {
-    fields = [
-      ...fields,
```

Comment:
Was this previous formulation creating a new array and copying it every time? If so, new formulation seems much better 👍